### PR TITLE
clean up the CoMID module

### PR DESCRIPTION
**Pull request by thomas-fossati**
As discussed at the last meeting, this patch brings the CoMID CDDL back in to "successfully validated" status

### CODE REVIEW

File: cddl/concise-mid-tag.cddl [hunk 1]
```diff
@@ -139,97 +151,113 @@ $device-id-type /= tagged-ueid
...
+; - vendor is the namespace authority designation
+; - type SHOULD include a namespace authority if vendor is ommitted
+; - An extension point in element-name confuses verifiers' ability to match
+;   against evidence
+; - label is defined in coswid schema (q: should we use a different value if
+;   the semantics differ?)
+; - type is defined in coswid schema
+element-name = {
+  ? label => text,
+  ? vendor => text,
```
Comment (henkbirkholz) [inline cddl/concise-mid-tag.cddl:193]: From the top of the head: is this (e.g. vendor) not a conflict with another cddl fragment?

File: cddl/concise-mid-tag.cddl [hunk 2]
```diff
@@ -139,97 +151,113 @@ $device-id-type /= tagged-ueid
...
+; Notes:
+; - vendor is the namespace authority designation
+; - type SHOULD include a namespace authority if vendor is ommitted
+; - An extension point in element-name confuses verifiers' ability to match
+;   against evidence
+; - label is defined in coswid schema (q: should we use a different value if
+;   the semantics differ?)
+; - type is defined in coswid schema
+element-name = {
+  ? label => text,
```
Comment (henkbirkholz) [inline cddl/concise-mid-tag.cddl:192]: this (e.g. label) is definitely defined somewhere else :-) do we want to create our own scope specific types here, e.g. "element-label"

File: cddl/concise-mid-tag.cddl [hunk 3]
```diff
@@ -139,97 +151,113 @@ $device-id-type /= tagged-ueid
...
 
-; NOTE(tho) from a syntactic PoV reference-value and endorsed-value are extremely similar.  I'm wondering why do we need to keep them separate?
-; NOTE(henk) I very much hope that they are similar: endorsements are on the "evidence side" of things as they kick in when the Attester cannot create evidence about the characteristics of its own components (roots of trust). Reference-values are the things that you compare the "evidence side" with. So endorsements cannot be part of the "reference side". They have to look similar/identical though - otherwise the comparison becomes complex and brittle.
-
-instance-claim-entry = {
-  ? element-name,
-  ; TLVs where instance-type identifies the data type e.g. MAC address, serial# etc.
-  instance-type => instance-values / [2* instance-values],
+instance-value = {
+  ? 0 => element-name
```
Comment (henkbirkholz) [inline cddl/concise-mid-tag.cddl:155]: Yeah... 

? element-name => element-name-entry

(+ renaming existing element-name)
(+ element-name = 0)

File: cddl/concise-mid-tag.cddl [hunk 4]
```diff
@@ -139,97 +151,113 @@ $device-id-type /= tagged-ueid
...
-layer=112
-index=113
-
-element-value = (
+; Notes:
+; - vendor is the namespace authority designation
+; - type SHOULD include a namespace authority if vendor is ommitted
+; - An extension point in element-name confuses verifiers' ability to match
+;   against evidence
+; - label is defined in coswid schema (q: should we use a different value if
```
Comment (henkbirkholz) [inline cddl/concise-mid-tag.cddl:188]: yes. in all cases, see below

File: cddl/concise-mid-tag.cddl [hunk 5]
```diff
@@ -139,97 +151,113 @@ $device-id-type /= tagged-ueid
...
+}
+
+vendor = 110
+model = 111
+layer = 112
+index = 113
+
+; Notes:
+; - do we need an extension point in element-value given there are extension
+;   points in endorsed-value and reference-value?
```
Comment (henkbirkholz) [inline cddl/concise-mid-tag.cddl:207]: sounds legit

File: cddl/concise-mid-tag.cddl [hunk 6]
```diff
@@ -139,97 +151,113 @@ $device-id-type /= tagged-ueid
...
+}
+
+version = 0
+svn = 1
+digests = 2
+flags = 3
+
+raw-value-entry = (
+  raw-value => bstr,
+  ? raw-value-mask => bstr,
```
Comment (henkbirkholz) [inline cddl/concise-mid-tag.cddl:223]: bitmask? there is potentially useful cddl for that. useful?

File: cddl/concise-mid-tag.cddl [hunk 7]
```diff
@@ -139,97 +151,113 @@ $device-id-type /= tagged-ueid
...
-digests=116
-flags=117
 
+raw-value = 4
+raw-value-mask = 5
+
+; Notes:
+; - one endorser benefits from the element-value of another endorser
+; - any-attribute is defined in CoSWID (this can be removed if we opt for using
+;   the extension point instead. NOTE(tho): agree
```
Comment (henkbirkholz) [inline cddl/concise-mid-tag.cddl:232]: Probably we already agreed on extension points (sockets) and actually no any-attributes at all in this scope? Have to reconfirm.

**Review by henkbirkholz — COMMENTED** [review]
looks good, in general
some comments in-line

File: cddl/concise-mid-tag.cddl (hunk 6 above)
Comment (nedmsmith) [reply cddl/concise-mid-tag.cddl:223]: Henk can you open a PR that has the bitmask CDDL you envisage?

File: cddl/concise-mid-tag.cddl [hunk 8]
```diff
@@ -139,97 +151,113 @@ $device-id-type /= tagged-ueid
...
+; - How do vendor specific schemas allocate indices (keys)? We need to specify
+;   a range of keys with rules 'first come first served'
 endorsed-value = {
-  element-name,
-  ? element-value, ; one endorser benefits from the element-value of another endorser
-  ? any-attribute, ; defined in coswid schema - **note: this can be removed if we opt for using the extension point instead. ** NOTE(tho): agree
-  * $$endorsed-value-extension ;How do vendor specific schemas allocate indices (keys)? We need to specify a range of keys with rules 'first come first served'
+  0 => element-name,
+  ? 1 => element-value,
+  ? any-attribute,
```
Comment (nedmsmith) [inline cddl/concise-mid-tag.cddl:238]: Need to remove any-attribute and only allow extension point extensibility

File: cddl/concise-mid-tag.cddl (hunk 2 above)
Comment (nedmsmith) [reply cddl/concise-mid-tag.cddl:192]: It feels awkward to have element repetition as in element-name.element-label although you end up with similar sort of repetition if you defined every field as a type and use a scalar RHS code point as in element-name = { 0 => element-label-type, ...} ... element-label-type /= text. 

Some of this feels like stylistic differences, maybe we just need to adopt a style guideline and be consistent?

**Pull request merged**